#! /bin/sh -e

# Description: java support for GNU/Hurd
# Author: Robert Millan, Petr Salinger, Samuel Thibault

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p1 < $0
        #cd ${dir}gcc && autoconf2.59
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p1 < $0
        #rm ${dir}gcc/configure
        ;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1
esac
exit 0


--- gcc-4.1.1/libjava/java/io/natFilePosix.cc.orig	2003-09-12 03:08:18.000000000 +0200
+++ gcc-4.1.1/libjava/java/io/natFilePosix.cc	2007-01-25 23:23:42.000000000 +0100
@@ -25,6 +25,7 @@
 #endif
 #include <string.h>
 #include <utime.h>
+#include <limits.h>
 
 #include <gcj/cni.h>
 #include <jvm.h>
@@ -379,6 +380,10 @@
 void
 java::io::File::init_native ()
 {
+#ifdef MAXPATHLEN
   maxPathLen = MAXPATHLEN;
+#else
+  maxPathLen = INT_MAX;
+#endif
   caseSensitive = true;
 }
diff -ur gcj-4.1.1/libjava/configure.host gcj-4.1.1/libjava/configure.host  
--- gcj-4.1.1/libjava/configure.host	2006-09-07 01:00:37.000000000 +0200
+++ gcj-4.1.1/libjava/configure.host	2006-09-07 01:00:37.000000000 +0200
@@ -326,7 +326,7 @@
 esac
 
 case "${host}" in
-  *linux*|*-kfreebsd*-gnu)
+  *linux*|*-kfreebsd*-gnu|*gnu*)
     use_libgcj_bc=yes
   ;;
   *)
